cli/src/wallet.rs

```diff
@@ -70,24 +74,47 @@ pub async fn new_wallet(cli: WalletCli) -> Result<(Option<Wallet>, Option<String
                 WalletCommand::Init(_) | WalletCommand::Mnemonic | WalletCommand::Restore { .. } => unreachable!(),
             };
 
-            (wallet, account)
+            (Some(wallet), account)
+        }
+    } else if storage_exists && snapshot_exists {
+        let wallet = Wallet::builder()
+            .with_secret_manager(secret_manager)
+            .with_storage_path(&storage_path)
+            .finish()
+            .await?;
+        if wallet.get_accounts().await?.is_empty() {
+            // ask the new user whether a default account should be created
+            if get_decision("Initialize a default account?")? {
+                println_log_info!("Initializing default account.");
+                let account = new_command(&wallet, None).await?;
+                (Some(wallet), Some(account))
+            } else {
+                (Some(wallet), None)
+            }
+        } else if let Some(account_index) = pick_account(&wallet).await? {
+            let accounts = wallet.get_accounts().await?;
+            let account = accounts[account_index as usize].alias().await;
```

Comment:
This should also work
```suggestion
            let account = wallet.get_account(account_index).await?;
```

Reply:
Actually `pick_account()` can return the account handle directly I think